test/tools/yulopti.cpp

```diff
@@ -171,28 +206,138 @@ class YulOpti
 		}
 	}
 
-	void disambiguate()
+	void applyFunctionToObject(shared_ptr<Object> _object, function<void(Object&)> _function)
 	{
-		*m_ast = std::get<yul::Block>(Disambiguator(m_dialect, *m_analysisInfo)(*m_ast));
-		m_analysisInfo.reset();
-		m_nameDispenser.reset(*m_ast);
+		for (auto const& subObjectNode: _object->subObjects) {
+			auto subObject = dynamic_pointer_cast<Object>(subObjectNode);
+
+			if (subObject != nullptr)
+				applyFunctionToObject(subObject, _function);
+		}
+
+		_function(*_object);
 	}
 
-	void runSteps(string _source, string _steps)
+	void runCodeAnalyzer(ErrorReporter& _errorReporter)
 	{
-		parse(_source);
-		disambiguate();
-		OptimiserSuite{m_context}.runSequence(_steps, *m_ast);
-		cout << AsmPrinter{m_dialect}(*m_ast) << endl;
+		applyFunctionToObject(
+			m_object,
+			[&](Object& _object)
+			{
+				_object.analysisInfo = make_shared<yul::AsmAnalysisInfo>();
+
+				AsmAnalyzer analyzer(
+					*_object.analysisInfo,
+					_errorReporter,
+					m_dialect,
+					{},
+					_object.qualifiedDataNames()
+				);
+
+				bool success = analyzer.analyze(*_object.code);
+				yulAssert(success && !_errorReporter.hasErrors(), "Invalid assembly/yul code.");
+			}
+		);
 	}
 
-	void runInteractive(string _source, bool _disambiguated = false)
+	void runCodeDisambiguator()
 	{
+		applyFunctionToObject(
+			m_object,
+			[&](Object& _object)
+			{
+				_object.code = make_shared<yul::Block>(
+					get<yul::Block>(Disambiguator(m_dialect, *_object.analysisInfo)(*_object.code))
+				);
+
+				_object.analysisInfo.reset();
+			}
+		);
+	}
+
+	void runSequence(string_view _steps)
+	{
+		applyFunctionToObject(
+			m_object,
+			[&](Object& _object)
+			{
+				OptimiserSuite{*m_context}.runSequence(_steps, *_object.code);
+			}
+		);
+	}
+
+	void runVarNameCleaner()
+	{
+		applyFunctionToObject(
+			m_object,
+			[&](Object& _object)
+			{
+				VarNameCleaner::run(*m_context, *_object.code);
+			}
+		);
+	}
+
+	void runStackCompressor()
+	{
+		applyFunctionToObject(
+			m_object,
+			[&](Object& _object)
+			{
+				StackCompressor::run(m_dialect, _object, true, 16);
+			}
+		);
+	}
+
+	void parseAndPrint(string _source, string _objectPath)
+	{
+		parse(_source, _objectPath);
+		printObject();
+	}
+
+	void printObject()
+	{
+		if (!m_inputWasCodeBlock)
+			cout << m_object->toString(&m_dialect) << endl;
+		else
+			cout << AsmPrinter{m_dialect}(*m_object->code)  << endl;
+	}
+
+	void resetNameDispenser()
+	{
+		m_nameDispenser = make_shared<NameDispenser>(
+			m_dialect,
+			m_reservedIdentifiers
+		);
+
+		m_context = make_shared<OptimiserStepContext>(
+			OptimiserStepContext{
+				m_dialect,
+				*m_nameDispenser,
+				m_reservedIdentifiers,
+				solidity::frontend::OptimiserSettings::standard().expectedExecutionsPerDeployment
+			}
+		);
+	}
+
+	void runSteps(string _source, string _objectPath, string _steps)
```

Comment:
Now that you added `runSequence()`, both sound like they would do the same thing. Let's make it less ambiguous:
```suggestion
	void parseAndRunSteps(string _source, string _objectPath, string _steps)
```